tests/jerry/func-decl.js

```diff
@@ -38,3 +38,18 @@ check_syntax_error ("'use strict'; function arguments () {}");
 check_syntax_error ("'use strict'; var l = function arguments () {}");
 
 check_syntax_error ("function f__strict_mode_duplicate_parameters (p, p) { 'use strict'; }");
+
+function test_strict_mode_propagation_in_func_expr_and_getters_setters () {
```

Comment:
Seems that I have misunderstood what test case do you ask for.
The test in the PR fails without the patch, and passes with the patch.
Issue, solved by the patch, is incorrect propagation of strict mode to outer scopes.

Yes, it is possible to construct test for `fail` folder. Could you, please, describe why do we need it?